### PR TITLE
Fix for #42: Deleted duplicate class and fixed occurrences of devon-java to devon4j and the property 'pluginName' in pom.xml

### PR DESCRIPTION
**Pull request by lmarniazman**
Fix for #42

### CODE REVIEW

File: src/main/java/com/devonfw/ide/sonarqube/common/impl/DevonPlugin.java [hunk 1]
```diff
@@ -1,17 +0,0 @@
...
-/***
- * {@link Plugin} that registeres the Devon rules in SonarQube.
- */
-public class DevonPlugin implements Plugin {
-
-  @Override
-  public void define(Context context) {
-
-    context.addExtension(DevonSonarDefinition.class);
-    context.addExtension(DevonSonarRegistrar.class);
```
Comment (hohwille) [inline src/main/java/com/devonfw/ide/sonarqube/common/impl/DevonPlugin.java]: IMHO we need to put these two lines into the other plugin class so we do not loose this functionality.

Comment (hohwille) [reply src/main/java/com/devonfw/ide/sonarqube/common/impl/DevonPlugin.java]: IMHO we should also rename the `DevonSonarPlugin` class to `SonarDevon4jPlugin`. Is that class referenced in our POM? Then we would need to update there as well.

Comment (lmarniazman) [reply src/main/java/com/devonfw/ide/sonarqube/common/impl/DevonPlugin.java]: The functionality won't be lost as those two classes are already added as an extension in the DevonSonarPlugin class. The renaming of the class is a good idea though, I will fix that asap.

**Comment by hohwille** [comment]
```
[INFO] User cache: /home/travis/.sonar/cache
[ERROR] SonarQube server [http://localhost:9000] can not be reached
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
...
[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.4.0.905:sonar (default-cli) on project sonar-devon4j-plugin: Unable to execute SonarQube: Fail to get bootstrap index from server: Failed to connect to localhost/127.0.0.1:9000: Connection refused (Connection refused) -> [Help 1]
```

Any ideas? Is `sonar-maven-plugin` starting up a server at port 9000? I was not aware of that. Or does it try to deploy the build plugin to the local running SonarQube server - so a development option is active that should be enabled only locally but never committed to git?

**Comment by hohwille** [comment]
As the build error in travis does not seem to relate to the changes done in this PR (they all look good to me), I simply restarted the travis build.
If if magically gets green now, I will simply merge. Otherwise we need to investigate what is going wrong here...

**Comment by hohwille** [comment]
Exactly the same error:
```
INFO] User cache: /home/travis/.sonar/cache
[ERROR] SonarQube server [http://localhost:9000] can not be reached
```

@lmarniazman  could you be so kind and run `mvn clean install` locally on your feature branch (`42-project_cleanup`)? If you get the same error, try some more investigation (`mvn clean install -X`) to trace down the issue (maybe try to revert parts of the changes and see which minimum change is causing the error). Also we may just run it on master and see if we can reproduce the error there.

**Comment by hohwille** [comment]
I just realized, that your other PR is causing the same error in travis.
Can you check what happens locally?
Can you check the POM and see if we have something there that starts a sonar qube or tries to connect to a local SonarQube that can not work in travis? We are kind of stuck or blocked until we have been able to fix travis and make it green again.

**Comment by lmarniazman** [comment]
Locally all branches build fine, which is very strange. I haven't gotten around to look into this error, but I will now

**Comment by lmarniazman** [comment]
See Issue #46, to be fixed by #47

**Pull request merged**